app/controllers/dinners_controller.rb

```diff
@@ -6,16 +6,21 @@ def new
   def create
     @dinner = Dinner.new(dinner_params)
     if @dinner.save
-      # TODO -- Flash a success
+      flash[:notice] = "Copy your link and send to friends: #{@dinner.custom_url}"
       redirect_to root_path
     else
+      flash.now[:error] = "Please complete all fields"
       render 'new'
```

Comment:
Hmm, no need to set a `flash` message here.  The `new` template should look for errors on the `@dinner` object and render them accordingly.  `form_for` is sensitive to these and will modify the HTML it generates to add special classes/wrappers to fields that have errors on them, making it possible to do things like

```css
.field_with_errors input {
  border: 1px solid red;
}
```

I forget the exact CSS classes/wrappers `form_for` adds.